base_rest_demo/tests/test_new_api.py

```diff
@@ -0,0 +1,19 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from .common import NewServiceCommonCase
+
+
+class TestPartner(NewServiceCommonCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.service = cls._get_service("partner")
+
+    def test_search(self):
+        self.env["res.partner"].create([{"name": f"P{idx}"} for idx in range(0, 1000)])
```

Comment:
Since the problem is not on the service layer, theses tests should be done into datamodels addon and should only involve the creation if a batch of model instances..

Reply:
Done